Server/ConcordiaCurriculumManager/Services/CourseGroupingService.cs

```diff
@@ -222,4 +223,28 @@ public async Task DeleteCourseGroupingRequest(Guid dossierId, Guid requestId)
             throw new ServiceUnavailableException("The course grouping could not be deleted");
         }
     }
+
+    public async Task<CourseGrouping> PublishCourseGrouping(Guid commonIdentifier)
+    {
+        var newCourseGrouping = await _courseGroupingRepository.GetCourseGroupingByCommonIdentifier(commonIdentifier) ?? throw new NotFoundException($"The course grouping with common ID {commonIdentifier} was not found.");
+        var oldCourseGrouping = await _courseGroupingRepository.GetPublishedVersion(commonIdentifier);
+
+        if (newCourseGrouping.Published)
+        {
+            return newCourseGrouping;
+        }
+
+        newCourseGrouping.MarkAsPublished();
+
+        if (oldCourseGrouping is not null)
+        {
+            _logger.LogInformation($"The course grouping with common ID {commonIdentifier} does not have an old published course grouping. This could be the first time it is published");
+            oldCourseGrouping.MarkAsUnpublished();
+            await _courseGroupingRepository.UpdateCourseGrouping(oldCourseGrouping);
+        }
+
+        await _courseGroupingRepository.UpdateCourseGrouping(newCourseGrouping);
+
```

Comment:
Can you log that we've published a grouping?

Reply:
Done